staging_vespalib/src/vespa/vespalib/metrics/current_samples.cpp

```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#include "current_samples.h"
+
+namespace vespalib {
+namespace metrics {
+
+void swap(CurrentSamples& a, CurrentSamples& b)
+{
+    using std::swap;
+    swap(a.counterIncrements, b.counterIncrements);
+    swap(a.gaugeMeasurements, b.gaugeMeasurements);
+}
```

Comment:
> Since add and sample are both mutex-protected, does swap need to take the mutexes
> of its operands too (in a well-defined order)? 

I tried that first, but it felt wrong to do it on such a low level, plus swap() can't know which mutex to take (and therefore must take both).  Now I take the mutex just around calling swap instead.  Longer-term I want to rewrite this with a better abstraction.